### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by danifromecuador**
## In this pull request my partner @Mylo16 and I have created a movie web app using an API and following the following requirements:
- Use ES6, linters, git flow, and Webpack
- Send and receive data from two different APIs
- Working with asynchronous Javascript
- Performing code reviews for team members
- Writting units test for JavaScript functions.

Also, we have recorded a video presentation:
- https://drive.google.com/file/d/1Nb_6EsuK566uN8HR-VdVr0pifxITkRaI/view

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  <header>
+    <nav>
+      <div class="logo-container"><img class="logo" src="../assets/logo.png" alt="application logo"></div>
+      <ul id="nav-lists">
+        <li class="nav-list"><a href="#arrow">Arrow(1)</a></li>
+        <li class="nav-list"><a href="#blacklist">BlackList(1)</a></li>
+        <li class="nav-list"><a href="#blacklist">The Last of Us(1)</a></li>
+      </ul>
+    </nav>
+  </header>
```
Comment (addod19) [inline src/index.html]: - [ ] It would be better if the home page header can be seen as we have in the mock up. Currently the header is not accessible as we can see in the screenshot below as compared to the mockup
Screenshot from 2023-04-22 22-01-50
Screenshot from 2023-04-22 21-59-54

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <main id="main">
+
+  </main>
+  <section id="popup-window">
+
+  </section>
+  <footer>
+    <p>Created by Eric and Dani under CC license</p>
+  </footer>
+  <!-- <script src="index.js" type="module"></script> -->
```
Comment (addod19) [inline src/index.html]: ### Optional
- [ ] It would be nice if we remove commented piece of code, it makes our codebase messy and difficult to understand

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+const mainBody = document.getElementById('main');
+const popup = document.getElementById('popup-window');
+
+window.addEventListener('load', async () => {
+  const numberOfLikes = await getLikes();
+  const epiArray = await apiGet();
+  for (let i = 0; i < epiArray.length; i += 1) {
+    if (epiArray[i].season === 1) {
+      mainBody.innerHTML += generateMovies(epiArray[i], numberOfLikes[i].likes);
```
Comment (addod19) [inline src/index.js:19]: - [ ] It would be better if we can see the like icons as we have in the mockup, our is not showing
Screenshot from 2023-04-22 22-07-39

File: src/modules/userInteraction.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+function generatePopup(info) {
+  return `
+    <div id = "modal">
+        <div id = "modal-content">
+            <img id = "close-btn" src = "../assets/closebtn.png" alt = "close button"/>
```
Comment (addod19) [inline src/modules/userInteraction.js]: - [ ] The close icon in the popup window is not showing as expected, it could be a path problem. Let us kindly resolve this

File: src/styles.css [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+  height: 2px;
+}
+
+/***** MAIN SECTION *****/
+
+#main {
+  width: 80%;
+  position: relative;
+  margin: auto;
+  display: flex;
```
Comment (addod19) [inline src/styles.css]: - [x] It would be better if we follow CSS best practices, like for example; `display: flex` has appeared more than 3 times so it would be better if we create a `class` for it and attach it to the HTML elements

File: src/styles.css [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+.nav-list:hover::after {
+  width: 100%;
+  height: 2px;
+}
+
+/***** MAIN SECTION *****/
+
+#main {
```
Comment (addod19) [inline src/styles.css]: - [x] It would be better if we use `class attributes` for styling our elements instead of the `ID` we have currently

File: README.md [hunk 7]
```diff
@@ -1,2 +1,93 @@
...
+
+If you like this project show your support by giving a ⭐️
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+We would like to thank the entire microverse staff and students for such a wonderful coding camp
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (addod19) [inline README.md:93]: - [x] It would be better if we create the MIT license file in our root directory so that we can fix the `404` error page not found

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @danifromecuador and @Mylo16, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Involvement API is used to record user interaction :rocket: 
- [ ] App contains a home page and a comments pop up :100: 
- [ ] The web app should retrieve data from the selected API and show the list of items on the screen :fireworks: 
- [ ] Homepage footer is similar to the mockup :+1: 
- [ ] Comment pop up works as expected :1st_place_medal: 
- [ ] When a user clicks on the Like button of an item, the interaction gets recorded in the Involvement API, and the screen gets updated :rocket: 
- [ ] Function for the counters present :+1: 
- [ ] All unit test passed with edge cases :100: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Capitalizing all commit messages and also write commit messages like `Fix` instead of `Fixed` going forward

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (Owusu-Desmond) [inline src/index.js:19]: - [ ] As requested by the previous code reviewer,  It would be better if you make the like icons show as we have in the mockup or [design guidelines](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md). Currently, the icons are not showing as you can see below :point_down: 

  image
  
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @danifromecuador and @Mylo16 :wave: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some changes you still need to work on before you move on to the next project but you are almost there.



 ## Highlights :+1: 
:heavy_check_mark: Good commit history.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Your project is well documented


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @danifromecuador 

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Descriptive PR ✔️
- Nice use of code semantics✔️
- Beautiful UI ✔️

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).
